knowledge-base/item-template-selector-collectionview-dotnet-maui.md

```diff
@@ -0,0 +1,90 @@
+
+## Description
+
+I am migrating from `RadListView` to [CollectionView](https://docs.telerik.com/devtools/maui/controls/collectionview/overview) for .NET MAUI. In `RadListView`, I used the `RadListView.ItemTemplateSelector` to dynamically select item templates at runtime. However, I could not find an equivalent feature for the CollectionView. How can I achieve dynamic item template selection in the CollectionView?
+
+This knowledge base article also answers the following questions:
+- How to use `ItemTemplateSelector` in CollectionView for .NET MAUI?
+- How to dynamically select templates in CollectionView?
+- Implementing custom `DataTemplateSelector` for CollectionView?
+
+## Solution
+
+To achieve dynamic item template selection in the CollectionView, you can use the `DataTemplateSelector` provided by .NET MAUI. The `DataTemplateSelector` enables you to define logic for choosing different templates based on item properties.
+
+### Steps to Implement ItemTemplateSelector:
+
+**1.** Create a Custom Template Selector Class:
+   Define a class that inherits from `DataTemplateSelector`. Override the `OnSelectTemplate` method to provide logic for selecting the appropriate template.
+
+```csharp
+   using Microsoft.Maui.Controls;
+
+   public class CustomTemplateSelector : DataTemplateSelector
+   {
+       public DataTemplate TemplateOne { get; set; }
+       public DataTemplate TemplateTwo { get; set; }
+
+       protected override DataTemplate OnSelectTemplate(object item, BindableObject container)
+       {
+           var myItem = item as MyItemType;
+           if (myItem != null && myItem.Condition)
+           {
+               return TemplateOne;
+           }
+           return TemplateTwo;
+       }
+   }
+```
+
+**2.** Define DataTemplates in XAML:
+   Declare the templates and the custom selector in your XAML file.
+
+```xml
+   <ResourceDictionary>
+       <DataTemplate x:Key="TemplateOne">
+           <Label Text="{Binding PropertyOne}" />
+       </DataTemplate>
+       <DataTemplate x:Key="TemplateTwo">
+           <Label Text="{Binding PropertyTwo}" />
+       </DataTemplate>
+
+       <local:CustomTemplateSelector x:Key="CustomSelector" 
+                                     TemplateOne="{StaticResource TemplateOne}" 
+                                     TemplateTwo="{StaticResource TemplateTwo}" />
+   </ResourceDictionary>
+```
+
+**3.** Apply the Template Selector to the CollectionView:
+   Set the `ItemTemplate` property of the CollectionView to the custom template selector.
+
+```xml
+<telerik:RadCollectionView ItemsSource="{Binding Items}" 
+                                                 ItemTemplate="{StaticResource CustomSelector}" />
+```
+
+### Additional Notes
+
+- Ensure your `ItemsSource` and item type are properly bound to the CollectionView.
+- Customize the logic in the `OnSelectTemplate` method based on your specific requirements.
```

Comment:
Isn't this stating the obvious (again). If you decide to keep it, move it to the first step where OnSelectTemplate is mentioned first.